worker/rhpv2.go

```diff
@@ -155,6 +160,65 @@ type Session struct {
 	mu          sync.Mutex
 }
 
+// transport wraps rhpv2.Transport to make sure we always apply the context's
+// deadline as read/write deadline to the transport's underlying connection.
+type transport struct {
```

Comment:
Are you sure this is better? `SetDeadline` does not work that way because `ctx.Deadline()` returns the same end time regardless of when you call it, so if you call it multiple times the deadline won't be further into the future

Reply:
Kept it for now

Reply:
I really don't care too much about what it's called. This is just an example. It can also be a `deadlineTransport` that is created with `newTransportWithDeadline` or whatever really.
It's just a lot less verbose because you don't need to wrap every method which is quite nasty.

Reply:
It was not about naming. I wanted to wrap the transport to ensure nobody would ever use the transport without setting the deadline. I rewrote though and I think I implemented what you were suggesting.